### ISSUE TITLE
Podspec with vendored framework - Headers not found - Could not build objective-c module

### ISSUE THREAD

**Issue by ldi13** [CocoaPods/CocoaPods#4135]
Hi CocoaPods team,

Currently, I'm facing an issue regarding headers contained in a vendored framework I use into an Objective-C podspec I'm creating. This podspec is supposed to be used in my main project as framework, because I declared "use_frameworks!" in my main project podfile.

Here is my podspec:

![capture d ecran 2015-09-07 a 15 42 42](https://cloud.githubusercontent.com/assets/11734827/9717925/29d30112-5577-11e5-80ef-8fe0b6808d58.png)

If I launch "pod lib lint --verbose --allow-warnings" to ensure that my podspec is valid, the build I obtained is successful.

However, after making "pod install" into the main project declaring my podspec, when I start the build, here is what I obtained:

![capture d ecran 2015-09-07 a 15 28 41](https://cloud.githubusercontent.com/assets/11734827/9717782/b787e042-5575-11e5-94cf-d53c60802ff6.png)

![capture d ecran 2015-09-07 a 15 29 01](https://cloud.githubusercontent.com/assets/11734827/9717839/36f4d5a6-5576-11e5-8024-bf58bb3a0541.png)

I followed the guidelines decribed in Pod Authors Guide to CocoaPods Frameworks, written by Marius Rackwitz, and I took care of declaring my header with < > and not " " (as first screenshot shows it).

I also tried to defined private, public headers, without any success, removed DerivedData directory, clean the project, etc...

This for sure a known issue, but I can't see what I'm missing here. 

Thanks for your support,

Lorenzo.

**Comment by segiddins** [comment]
Can you share a project we can use to run, as without that it's very hard to guess what might be wrong. What version of CocoaPods are you using?

**Comment by neonichu** [comment]
Could you please try if adding

```ruby
s.ios.public_header_files  = 'CastlabsSDK.framework/Versions/A/Headers/*.h'
```

changes anything?

**Comment by ldi13** [comment]
Hello Team, 

Thank you very much for your answers.

@segiddins I'm not sure that I can share the project on which I have problems. The podspec I'm using is deployed on a private repository, or locally, may be I could share both. Do you have an email ? 
The cocoapods current version I'm using is 0.38.2.

@neonichu I tried the following already: 
s.ios.public_header_files  = 'CastlabsSDK.framework/Headers/*.h' 

When running pod install on the main project, I obtain the following error, mentioned in Marius Rackwitz 
Include of non-modular header inside framework module

![capture d ecran 2015-09-07 a 17 15 17](https://cloud.githubusercontent.com/assets/11734827/9719395/29eaedec-5584-11e5-81b1-592fc84d6892.png)

Considering my CLPlayerWrapper, here is how my header is declared into it: 

![capture d ecran 2015-09-07 a 17 19 20](https://cloud.githubusercontent.com/assets/11734827/9719442/9c7c0e36-5584-11e5-85fc-83abb745f895.png)

**Comment by ldi13** [comment]
OK,
I fixed several things afterwards:

First one, to declare "use_frameworks!" in my Podfile main project.
This is how my podfile looks like:

![capture d ecran 2015-09-07 a 19 34 42](https://cloud.githubusercontent.com/assets/11734827/9721244/a55090c8-5597-11e5-9f1a-35c543e90ae0.png)


The error I got now is located on the Header-umbrella.h file generated:

![capture d ecran 2015-09-07 a 19 31 59](https://cloud.githubusercontent.com/assets/11734827/9721278/222178ce-5598-11e5-9bde-045c2fd5db7f.png)

**Comment by segiddins** [comment]
I'm moving this week, so I don't know if I'll be able to investigate for a few weeks. You'll have a much better shot of someone on the team getting to it if you post it here.

**Comment by ldi13** [comment]
OK, Thank you @segiddins.
I transfered the email you received to @neonichu.

**Comment by neonichu** [comment]
I took a look at the project you sent — since you have defined `source_files`, it appears that you have to add the public headers of the framework to both the `source_files` and the `public_header_files` attributes. I'm actually not sure if that's a bug or expected behaviour. @segiddins what do you think?

```ruby
s.source_files = 'Pod/Classes/**/*', 'CastlabsSDK.framework/Headers/*.h'
s.public_header_files = 'CastlabsSDK.framework/Headers/*.h'
```

In any case, after changing this, you run into the problem that your headers expose C++, so they're not usable from Swift, but that's unrelated to CocoaPods.

**Comment by segiddins** [comment]
@neonichu expected behavior. Public / private just acts as a filter.

**Comment by ldi13** [comment]
Hi guys, thanks for your answers.

@neonichu I agree with you, you cannot expose C++ into Swift code, you have to create Objective-C wrappers. That's what I have done into my code (CLPlayerWrapper).

Furthermore, The podspec is working correctly into my main project, if I remove "use_frameworks!" into the Podfile, and if I add a bridging header to declare my two headers contained into my podspec.

I cannot figure it out why I have building frameworks, if I want my podspec to be a dynamic framework.

**Comment by neonichu** [comment]
If you make the changes I suggested to your podspec, you should see that some public headers do use C++ directly.

**Comment by ldi13** [comment]
Yes @neonichu,

I made the changes you suggested. I removed C++ headers from public headers, and the building now is working, regarding dynamic frameworks. I still have an issue during "Copying swift docs phases". But it's completely independent regarding Cocoapods. I'm closing the issue though.

Once again, thank you so much for your support, you and @segiddins , I really appreciate it. Thank you for all the highlights you gave me regarding public headers into dynamic frameworks. 

Cocoapods is an amazing tool you're providing us.

**Comment by neonichu** [comment]
Glad we could help!

**Comment by DmIvanov** [comment]
Hi, guys! You're doing great job!

It seems like I run into the same problem, but your advices didn't help me.
I have my development pod (objC) with some dependent pods (objC). And I wanna use  my development pod in some swift project. As a result my pod's files don't see files from dependent pods - I have "file not found error". Tried to add `public_header_files` as it was written before, but the result was almost the same. (I've checked my pod with objC project - everything worked correctly - so the problem connected with main project written on swift).

I made a test project: https://github.com/DmIvanov/TEST.testPods

Thanks in advance!
